library/common.cpp

```diff
@@ -1,6 +1,32 @@
 #include "elliptics.h"
 
-size_t dnet_backend_info_list_count(dnet_backend_info_list *backends)
+int dnet_get_backend_ids(const dnet_backend_info_manager *backends, size_t **backend_ids, size_t *num_backend_ids)
 {
-	return backends ? backends->backends.size() : 0;
+	if (!backends)
+		return -EINVAL;
+
+	auto config_backends = backends->get_all_backends();
+	*num_backend_ids = config_backends.size();
+	*backend_ids = reinterpret_cast<size_t *>(malloc(*num_backend_ids * sizeof(size_t)));
+	if (!*backend_ids)
+		return -ENOMEM;
+
+	for (size_t i = 0; i < *num_backend_ids; ++i) {
+		(*backend_ids)[i] = config_backends[i]->backend_id;
+	}
+
+	return 0;
+}
+
+struct dnet_backend_io *dnet_get_backend_io(struct dnet_io *io, size_t backend_id)
```

Comment:
This function doesn't increase reference counter or perform any other control flow operations (like remove object from array) else, why does it use locking?